pandas/io/excel.py

```diff
@@ -270,6 +277,10 @@ def parse(self, sheetname=0, header=0, skiprows=None, skip_footer=0,
         if skipfooter is not None:
             skip_footer = skipfooter
 
+        # if has_index_names:
```

Comment:
Yeah, I'll deprecate.  There is a case where `has_index_names` will be necessary to read output saved in pre 0.17, I'll add a warning and example to the docs, along with a couple tests.